src/expr.rs

```diff
@@ -1779,7 +1779,10 @@ fn rewrite_call_args(context: &RewriteContext,
     };
 
     match write_list(&item_vec, &fmt) {
-        Some(ref s) if !s.contains('\n') && s.len() > one_line_width => {
+        // If arguments do not fit in a single line and do not contain newline,
+        // try to put it on the next line.
+        Some(ref s) if context.config.fn_call_style == IndentStyle::Block &&
+                       (!s.contains('\n') && s.len() > one_line_width) => {
```

Comment:
1. with functions, it is not an error to insert commas in general , but with macros it is, e.g.,

```
f(a_long_arg);
// formatted to
f(
    a_long_arg, // OK
);
m!(a_long_arg);
// formatted to
m!(
    a_long_arg, // could introduce a compile error or change semantics
);
```

The underlying reason here is that arguments to macros are not a list of expressions (as they are for functions), but they are a list of tokens, so a macro can potentially 'see' the commas and change behaviour accordingly.

2. This is a bug for macros, if there is not a trailing comma, then we should not add one under any circumstances.